### PR TITLE
Removing trailing space and '\r' for windows

### LINKED ISSUE #470: DMI plugin stripping issue
Hi everybody,

There is a slice issue when using the dmi plugins on windows the parsing of dmidecode doesn't strip the reponse so every value has a '\r'.

Regards,
JM

### PR DESCRIPTION
**Pull request by jmauro**
Closes #470

### CODE REVIEW

**Comment by lamont-granquist** [comment]
@chef/client-engineers @chef/client-core needs review

File: lib/ohai/common/dmi.rb [hunk 1]
```diff
@@ -107,15 +107,15 @@ def convenience_keys(dmi)
               next if field.to_s == 'record_id'
               translated = field.downcase.gsub(/[^a-z0-9]/, '_')
               if in_common.has_key?(translated)
-                in_common[translated] = nil unless in_common[translated] == value
+                in_common[translated] = nil unless in_common[translated] == value.strip
```
Comment (jdmundrawala) [inline lib/ohai/common/dmi.rb]: I think you should pull `value.strip` up since it is being repeated multiple times

**Comment by danielsdeleo** [comment]
Agree w/ @jdmundrawala :+1: otherwise

File: spec/unit/plugins/dmi_spec.rb [hunk 2]
```diff
@@ -91,6 +91,10 @@
 	Contained Elements: 0
 EOS
 
+def convert_window_output(stdout)
```
Comment (tyler-ball) [inline spec/unit/plugins/dmi_spec.rb]: This seems useful enough to put into a `spec_helper` of some kind - I would normally only advocate this if something else already needed this functionality, but I think putting it in a helper will encourage future spec writers to leverage it if they see it being used here.

**Comment by tyler-ball** [comment]
:+1: other than my minor comment and echoing Jay/Dan

**Comment by jmauro** [comment]
Done

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -32,6 +32,10 @@ def get_plugin(plugin, ohai = Ohai::System.new, path = PLUGIN_PATH)
   loader.load_plugin(File.join(path, "#{plugin}.rb"))
 end
 
+def convert_window_output(stdout)
```
Comment (btm) [inline spec/spec_helper.rb]: convert_windows_output?

Comment (jmauro) [inline spec/spec_helper.rb]: This wasn't what I was supposed to do?

Comment (danielsdeleo) [inline spec/spec_helper.rb]: I think @btm is just saying this should be "windows" with an "s" at the end, minor typo.

**Comment by jmauro** [comment]
Typo corrected.

**Comment by danielsdeleo** [comment]
All the code review stuff has been addressed and this has all the :+1:s it needs, let's merge it.

**Comment by tyler-ball** [comment]
appveyor tests failing - looks like the rename needs to be added in those tests

**Comment by jmauro** [comment]
Fixing the method calls into the spec.

**Comment by btm** [comment]
:+1:

**Comment by btm** [comment]
rebased into 55ae7f38627c6eab2599fb00fc59395b9d6a60e1 and merged to master.

**Pull request closed without merging**